pkg/cmd/codespace/create.go

```diff
@@ -118,7 +120,15 @@ func (a *App) Create(ctx context.Context, opts createOptions) error {
 		Location:   opts.location,
 	}
 
-	promptForRepoAndBranch := userInputs.Repository == ""
+	if opts.useWeb && (opts.displayName != "" || opts.idleTimeout != 0 || opts.retentionPeriod.Duration != nil) {
+		return errors.New("using --web with --display-name, --idle-timeout, or --retention-period is not yet supported")
+	}
```

Comment:
Instead of doing the mutual exclusion check inside the main command, the usual pattern in the cli is to either use the `cobra` library's helper:

```go
createCmd.Flags().BoolVarP(&opts.useWeb, "web", "w", false, "create codespace from browser, cannot be used with --display-name, --idle-timeout, or --retention-period")

if err := createCmd.MarkFlagsMutuallyExclusive("web", "display-name"); err != nil {
   return err;
}
// repeat for other options
```

or use the custom `cmdutil.MutuallyExclusive` helper

```go
createCmd := &cobra.Command{
		// ...
		RunE: func(cmd *cobra.Command, args []string) error {
			if err := cmdutil.MutuallyExclusive(
				"using --web with --display-name, --idle-timeout, or --retention-period is not yet supported",
				opts.useWeb,
				opts.displayName != "" || opts.idleTimeout != 0 || opts.retentionPeriod.Duration != nil,
			); err != nil {
				return err
			}

			return app.Create(cmd.Context(), opts)
		},
	}
```

I think in this case the resulting error is nicer using `cmdutil` as it will print the error message as well as show the `--help` text. Either way, it's better to put the mutual exclusion rules close to the flag definitions so that they're easier to see immediately

Reply:
fixed in [c7db18a](https://github.com/cli/cli/pull/7288/commits/c7db18a3858bc60dfa33a9fd3ddc86e93afe369c) .
took me awhile to figure out how to test this. thankfully I figure it out by imitating `pkg/cmd/repo/create/create_test.go` (it also use `cmdutil.MutuallyExculsive` fn in `runE` clause)

please let me know if I need to change it further.

Reply:
I also refactor the mutual exclusive flag check logic in `list.go` , I applied same change and add same test. 
It's on this [0798456](https://github.com/cli/cli/pull/7288/commits/079845623f0e7b52f3357992a102140b190c6deb) commit.

Please let me know what you think.